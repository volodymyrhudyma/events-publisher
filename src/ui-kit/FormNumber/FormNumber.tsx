import React, { ChangeEvent } from 'react';
import { IFormControlProps } from 'src/types';
import FormControl from 'src/ui-kit/FormControl';
import Input from 'src/ui-kit/Input';
import Label from 'src/ui-kit/Label';
import { TipWrapper, Wrapper } from './styled/FormNumber.styled';

interface IFormNumberProps extends IFormControlProps {
  placeholder: string;
  tip: string;
  value: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const FormNumber = ({
  placeholder,
  tip,
  value,
  onChange,
  ...rest
}: IFormNumberProps) => (
  <FormControl {...rest}>
    <Wrapper>
      <Input
        type="number"
        placeholder={placeholder}
        width={90}
        value={value}
        onChange={onChange}
      />
      <TipWrapper>
        <Label color="secondary" uppercase={false}>
          {tip}
        </Label>
      </TipWrapper>
    </Wrapper>
  </FormControl>
);

FormNumber.defaultProps = {
  required: false,
};

export default FormNumber;
