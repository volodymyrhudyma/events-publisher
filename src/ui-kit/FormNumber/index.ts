import Component from './FormNumber';

export default Component;
