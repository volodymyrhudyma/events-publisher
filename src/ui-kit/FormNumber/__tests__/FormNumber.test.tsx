import React from 'react';
import { shallow } from 'enzyme';
import FormNumber from '../FormNumber';

describe('FormNumber component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormNumber
        label="Label"
        placeholder="Placeholder"
        tip="Tip"
        value=""
        onChange={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
