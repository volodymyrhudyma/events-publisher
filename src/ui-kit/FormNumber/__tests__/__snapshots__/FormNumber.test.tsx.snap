// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormNumber component should render correctly 1`] = `
<FormControl
  label="Label"
  required={false}
>
  <styled.div>
    <Input
      onChange={[MockFunction]}
      placeholder="Placeholder"
      type="number"
      value=""
      width={90}
    />
    <styled.div>
      <Label
        color="secondary"
        uppercase={false}
      >
        Tip
      </Label>
    </styled.div>
  </styled.div>
</FormControl>
`;
